lib/utilities.js

```diff
@@ -329,6 +332,33 @@ const jsonParse = (buffer) => {
   }
 };
 
+const flatObject = (sourceObj, nameField) => {
+  const targetObj = {};
+
+  for (const [key, value] of Object.entries(sourceObj)) {
+    if (!isObject(value)) {
+      targetObj[key] = value;
+      continue;
+    }
+
+    if (nameField !== undefined && nameField !== key) {
+      targetObj[key] = { ...value };
+      continue;
+    }
+
+    for (const [key2, value2] of Object.entries(value)) {
+      const newKey = `${key}${toUpperCamel(key2)}`;
+
+      if (sourceObj[newKey] !== undefined)
+        throw new Error('flatObject: duplicate key');
+
+      targetObj[newKey] = value2;
```

Comment:
if here nested object you will copy a link
```javascript
  const source = {
    name: { first: 'Andrew', second: 'Johnson' },
    old: true,
    avoid: [1, 2, 3],
    parent: { mother: 'Eva', father: 'Adam', relatives: { stepBrother: 'Adam' } },
  };

 const expected = {
    nameFirst: 'Andrew', 
    nameSecond: 'Johnson' ,
    old: true,
    avoid: [1, 2, 3],
    parentMother: 'Eva', 
    parentFather: 'Adam', 
    parentRelatives: { stepBrother: 'Adam' },
  };

  source.parent.relatives.stepBrother = 'Tom';
  console.log(expected.parentRelatives.stepBrother); // 'Tom'
```
```suggestion
      targetObj[newKey] = isObject(value2) ? {...value2} : value2;
```
@AlexBelch